### ISSUE TITLE
can not remove project once if has been removed from the filesystem

### ISSUE THREAD

**Issue by yucer** [jccguimaraes/atom-project-viewer#81]
# ISSUE
I have deleted the project from my file system. Once I'm back in atom I want to delete it from the group of project viewer. 

## Steps
Please try to describe **step by step** how to get to the issue.

1. Remove a project folder from the file system
2. Go back to Project viewer and try to delete the project

## Expected result

* A confirmation appears: This project was already deleted from /path/to/the/project. Are you sure you want to delete it from the group?

* If the answer is "yes", then delete it from the project group. Otherwise mark it as red or something like that.

## Environment

**Atom Version**: 1.5.4
**System**: Ubuntu 16.04.1
**Thrown From**: [project-viewer](https://github.com/jccguimaraes/atom-project-viewer) package, v0.2.34

### Stack Trace

Uncaught TypeError: Cannot read property 'children' of undefined

```
At /home/yucer/.atom/packages/project-viewer/lib/views/groups-view.js:99

TypeError: Cannot read property 'children' of undefined
    at groups-view.removeProjectNode (/home/yucer/.atom/packages/project-viewer/lib/views/groups-view.js:99:38)
    at groups-view.updatedData (/home/yucer/.atom/packages/project-viewer/lib/views/groups-view.js:181:45)
    at Function.module.exports.Emitter.simpleDispatch (/usr/share/atom/resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/usr/share/atom/resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at remove-project-view.removeProject (/home/yucer/.atom/packages/project-viewer/lib/views/remove-project-view.js:57:31)
    at HTMLButtonElement.removeButtonClick (/home/yucer/.atom/packages/project-viewer/lib/views/remove-project-view.js:28:36)
```

### Commands

```
  2x -5:26.4.0 editor:select-to-end-of-word (atom-text-editor.editor.is-focused)
     -5:26.2.0 editor:select-to-beginning-of-word (atom-text-editor.editor.is-focused)
     -5:25.3.0 core:copy (atom-text-editor.editor.is-focused)
     -5:20.8.0 core:paste (atom-text-editor.editor.is-focused)
     -5:19.5.0 core:save (atom-text-editor.editor.is-focused)
  2x -3:51.8.0 editor:newline (atom-text-editor.editor.is-focused)
     -3:41.3.0 core:paste (atom-text-editor.editor.is-focused)
     -3:39.6.0 core:backspace (atom-text-editor.editor.is-focused)
     -3:38.3.0 core:save (atom-text-editor.editor.is-focused)
     -3:31.2.0 editor:select-to-end-of-word (atom-text-editor.editor.is-focused)
     -3:29.9.0 core:select-left (atom-text-editor.editor.is-focused)
     -3:29.1.0 core:backspace (atom-text-editor.editor.is-focused)
  2x -3:27.3.0 core:save (atom-text-editor.editor.is-focused)
     -2:15.5.0 project-viewer:toggle (atom-text-editor.editor.is-focused)
     -1:25 project-viewer:remove-group (span.)
  4x -1:12.3.0 project-viewer:remove-project (span.)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "split-diff"
    ],
    "themes": [
      "atom-light-ui",
      "atom-light-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
atom-beautify, v0.28.28
atom-html-preview, v0.1.22
hide-files, v0.1.7
language-docker, v1.1.7
language-gherkin-i18n, v1.3.4
language-restructuredtext, v1.0.0
odoo-snippets, v0.3.0
prettifier, v0.2.4
project-viewer, v0.2.34
python-tools, v0.6.8
rst-preview, v1.3.1
tablr, v1.3.2
tabs-to-spaces, v1.0.2
xml-tools, v0.2.1

# Dev
No dev packages
```

**Comment by jccguimaraes** [comment]
hi @yucer, thanks for reporting this but it seems that you are using a really old version `0.2.34`an at the moment it's on `0.3.20`.
Can you please update the package and see if the result is the same? Thanks

**Comment by yucer** [comment]
I didn't realize that. I always make this:

```
$ apm update
Package Updates Available (0)
└── (empty)
```

And it doesn't report me updates available. Should I manually uninstall 0.2 and install 0.3 ?

Would I loose my project list ? Can I export it  somehow ?

**Comment by jccguimaraes** [comment]
wow! that's odd.

Well i advise you to save it to be safe, it will convert it but don't know if from that late version it will do it correctly.

it's on the .atom folder, file called `project-viewer.json`. You don't need to copy it afterwards

**Comment by jccguimaraes** [comment]
@yucer any luck?

**Comment by jccguimaraes** [comment]
Regarding the update do you have latest atom version also? I usually target ro the latest

**Comment by yucer** [comment]
I did update atom and the project-viewer addon update was detected.

The project info was automatically converted. 

I did:

- added the new path to the project  
- remove the old path
- click on update button

the project started to be  shown deleted (with a line through). 

I did repeat the actions and everything was ok now.

The problem was that I have located all the projects of a group, I did change the location of the parent dir of all the projects, now I have to move the projects one by one. 

Thanks
